REFACTOR: move the 0% commands to `nu-cmd-extra`

amtoine opened the pull request:
requires
- https://github.com/nushell/nushell/pull/9455

# :gear: Description
in this PR i move the commands we've all agreed, in the core team, to move out of the core Nushell to the `extra` feature.

> **Warning**
> in the first commits here, i've
> - moved the implementations to `nu-cmd-extra`
> - removed the declaration of all the commands below from `nu-command`
> - made sure the commands were not available anymore with `cargo run -- -n`

## the list of commands to move
with the current command table downloaded as `commands.csv`, i've run
```bash
let commands = (
    open commands.csv
    | where is_plugin == "FALSE" and category != "deprecated"
    | select name category "approv. %"
    | rename name category approval
    | insert treated {|it| (
        ($it.approval == 100) or                # all the core team agreed on them
        ($it.name | str starts-with "bits") or  # see https://github.com/nushell/nushell/pull/9241
        ($it.name | str starts-with "dfr")      # see https://github.com/nushell/nushell/pull/9327
    )}
)
```
to preprocess them and then
```bash
$commands | where {|it| (not $it.treated) and ($it.approval == 0)}
```
to get all untreated commands with no approval, which gives
```
╭────┬───────────────┬─────────┬─────────────┬──────────╮
│  # │     name      │ treated │  category   │ approval │
├────┼───────────────┼─────────┼─────────────┼──────────┤
│  0 │ fmt           │ false   │ conversions │        0 │
│  1 │ each while    │ false   │ filters     │        0 │
│  2 │ roll          │ false   │ filters     │        0 │
│  3 │ roll down     │ false   │ filters     │        0 │
│  4 │ roll left     │ false   │ filters     │        0 │
│  5 │ roll right    │ false   │ filters     │        0 │
│  6 │ roll up       │ false   │ filters     │        0 │
│  7 │ rotate        │ false   │ filters     │        0 │
│  8 │ update cells  │ false   │ filters     │        0 │
│  9 │ decode hex    │ false   │ formats     │        0 │
│ 10 │ encode hex    │ false   │ formats     │        0 │
│ 11 │ from url      │ false   │ formats     │        0 │
│ 12 │ to html       │ false   │ formats     │        0 │
│ 13 │ ansi gradient │ false   │ platform    │        0 │
│ 14 │ ansi link     │ false   │ platform    │        0 │
│ 15 │ format        │ false   │ strings     │        0 │
╰────┴───────────────┴─────────┴─────────────┴──────────╯
```
# :paintbrush: User-Facing Changes
```
$nothing
```

# :test_tube: Tests + Formatting
- :black_circle: `toolkit fmt`
- :black_circle: `toolkit clippy`
- :black_circle: `toolkit test`
- :black_circle: `toolkit test stdlib`

# :book: After Submitting
```
$nothing
```

# :mag: For reviewers
```bash
$commands | where {|it| (not $it.treated) and ($it.approval == 0)} | each {|command|
    try {
        help $command.name | ignore
    } catch {|e|
        $"($command.name): ($e.msg)"
    }
}
```
should give no output in `cargo run --features extra -- -n` and a table with 16 lines in `cargo run -- -n`

stormasm commented:
@amtoine I thought we were going to move the *math* commands and the *bytes* commands first prior to considering doing these ?

That is why I started with bits...

In our core team meeting a few weeks back we all agreed to do those first...

So I will let others comment on which commands they want to move across besides
math and bytes which everyone has already signed off on 😄

stormasm commented:
However, I am OK with moving all of these commands across to nu-cmd-extra as well...

stormasm reviewed (approved):
I am fine with moving these commands across to nu-cmd-extra...
However we should get feedback from some other folks as well...
Just to make sure they are OK with it too. 😄

amtoine commented:
@stormasm 
i've asked this in the core team channel :relieved:

amtoine commented:
i'll continue working on this in the meantime, this won't be lost work :ok_hand:

amtoine commented:
@stormasm 
some news and improvements here :relieved: 
- i've solved the conflicts
- moved some functions and assets
- moved the missing `ansi link` command

now i'm trying to fix the tests but i have some trouble...
i was able to solve `help_works_with_missing_requirements` but now the examples of `each while` and `update cells` are failing and i do not understand what is the problem here :confused:

stormasm commented:
@amtoine see my comments to you in discord 😄

amtoine commented:
cc/ @stormasm 

the CI still is suspicious but i think this PR is somewhat unrelated to the CI issue.
as the jobs should all be :green_circle: as in amtoine/nushell#4, i think we can land this, if you're still ok with the changes :relieved:

sholderbach commented on crates/nu-cmd-extra/Cargo.toml:
```diff
@@ -18,6 +18,8 @@ nu-parser = { path = "../nu-parser", version = "0.82.1" }
 nu-protocol = { path = "../nu-protocol", version = "0.82.1" }
 nu-cmd-base = { path = "../nu-cmd-base", version = "0.82.1" }
 nu-utils = { path = "../nu-utils", version = "0.82.1" }
+nu-command = { path = "../nu-command", version = "0.82.1" }
+nu-cmd-lang = { path = "../nu-cmd-lang", version = "0.82.1" }
```
That is a nope! `nu-cmd-extra` should not regurlarly depend on them (only possibly as `dev-dependencies` if not circular)

amtoine replied:
sheeeesh, according to `git blame`, this has been introduced in #9543 through the merge commit :thinking:

amtoine replied:
it's like this on the `main` branch :eyes:

amtoine replied:
[here](https://github.com/nushell/nushell/blob/main/crates/nu-cmd-base/Cargo.toml#L13), do you want me to change this in this PR? :thinking:

sholderbach replied:
#9543 is innocent: https://github.com/nushell/nushell/pull/9543/files#diff-111d1354eee39d38926ac7554b9802115710a8e6f5122d904adb1dab403cccb9

I remember I made an effort to isolate this in #9429: https://github.com/nushell/nushell/pull/9429/files#diff-111d1354eee39d38926ac7554b9802115710a8e6f5122d904adb1dab403cccb9

amtoine replied:
i'm not sure i understand what should be done here if something needs to be addressed :confused:

sholderbach replied:
Those two lines should be gone (or only dev dependencies) and the crate should still compile.

amtoine replied:
okey, i see, let me try that :yum: 

(btw, it was introduced in #9327, my bad...)

amtoine replied:
it does fail with
```rust
error[E0433]: failed to resolve: use of undeclared crate or module `nu_command`
  --> crates/nu-cmd-extra/src/example_test.rs:64:43
   |
64 | ...dd_decl(Box::new(nu_command::Enumerate));
   |                     ^^^^^^^^^^ use of undeclared crate ormodule `nu_command`

For more information about this error, try `rustc --explain E0433`.
error: could not compile `nu-cmd-extra` due to previous error
``` :eyes:

amtoine replied:
iirc we need these to run the example tests of `each while` and `update cells` which require `enumerate` and `if` statement :thinking:

sholderbach replied:
Yeah but `example_test.rs` should be `#[cfg(test)]` only if I recall. So that would be fine as a dev dep.

sholderbach replied:
Thanks!

amtoine replied:
:partying_face:

stormasm reviewed (commented):
Great job @amtoine
